modules/core/shared/src/test/scala/scaladex/core/model/ArtifactTests.scala

```diff
@@ -141,13 +141,70 @@ class ArtifactTests extends AnyFunSpec with Matchers {
       obtained should contain(expected)
     }
   }
+
+  describe("scastieURL") {
+    it("should return None if artifact is SbtPlugin") {
+      createArtifact(
+        groupId = "com.typesafe.sbt",
+        artifactId = "sbt-native-packager_2.10_0.13",
+        version = "1.2.2",
+        binaryVersion = BinaryVersion(SbtPlugin.`0.13`, Scala.`2.10`)
+      ).scastieURL should be(None)
+    }
+
+    it("should return None if artifact is ScalaNative") {
+      createArtifact(
+        groupId = "org.typelevel",
+        artifactId = "cats-core",
+        version = "2.6.1",
+        binaryVersion = BinaryVersion(ScalaNative.`0.4`, Scala.`2.13`)
+      ).scastieURL should be(None)
+    }
+
+    it("should return None if artifact doesn't have ScalaVersion") {
+      createArtifact(
+        groupId = "org.typelevel",
+        artifactId = "",
+        version = "2.6.1",
+        binaryVersion = BinaryVersion(Jvm, Java)
+      ).scastieURL should be(None)
+    }
+
+    it("should return a valid URL") {
+      createArtifact(
+        groupId = "org.scalameta",
+        artifactId = "metals_2.13",
+        version = "0.11.8",
+        binaryVersion = BinaryVersion(Jvm, Scala.`2.13`),
+        projectRef = Some(Project.Reference.from("scalameta", "metals"))
+      ).scastieURL should be(
+        Some("https://scastie.scala-lang.org/try?g=org.scalameta&a=metals&v=0.11.8&o=scalameta&r=metals&t=JVM&sv=2.13")
+      )
+    }
+
+    it("should return a valid URL for a ScalaJs artifact") {
+      createArtifact(
+        groupId = "org.scala-js",
+        artifactId = "scalajs-library_2.13",
+        version = "1.16.0",
+        binaryVersion = BinaryVersion(ScalaJs.`1.x`, Scala.`3`),
+        projectRef = Some(Project.Reference.from("Scala.js", "scalajs-library"))
+      ).scastieURL should be(
+        Some(
+          "https://scastie.scala-lang.org/try?g=org.scala-js&a=scalajs-library&v=1.16.0&o=scala.js&r=scalajs-library&t=JVM&sv=2.13"
```

Comment:
This `t=JVM` looks wrong. If it is a Scala.Js library the target param should be `JS`. The problem is not in the implementation but in the `createArtifact` method:

Let's look at this line of code:
```scala
val artifactIdResult =
      artifactName.map(name => ArtifactId(name, binaryVersion)).orElse(Artifact.ArtifactId.parse(artifactId)).get
```

It can translate to: if there is an `artifactName` we combine it with `binaryVersion` to create the `ArtifactId` otherwise we parse the `String` in the `artifactId` argument. Whenever `artifactName` is `None` the `binaryVersion` is just ignored. In this test the `artifactName` is `None` so the given `binaryVersion` (`BinaryVersion(ScalaJs.1.x, Scala.3)`) is never used. Instead the `createArtifact` get the `binaryVersion` from parsing the `artifactId`. The `artifactId` is `scalajs-library_2.13` which means the binary version is `BinaryVersion(JVM, Scala.2.13)`. For a Scala.js artifact, the `artifactId` should be something like `scalajs-library_sjs1_2.13`.

Remark however that the real `scalajs-library_sjs1_2.13` artifact does not exist because `scalajs-library` is not a Scala.js library, but a JVM library. It's quite surprising. Instead you can use the `scalajs_dom_sjs1_2.13` artifact ([link](https://index.scala-lang.org/scala-js/scala-js-dom/artifacts/scalajs-dom/2.8.0?binary-version=_sjs1_2.13))

Reply:
I thought `scalajs-library` is a Scala.js library and built URL according to that. Thank you for clarification